src/codewhisperer/activation.ts

```diff
@@ -206,6 +209,11 @@ export async function activate(context: ExtContext): Promise<void> {
         )
     )
 
+    if (!isCloud9() && !auth.isConnectionValid()) {
+        // this is to proactively check and reflect the state if user's connection is expired
+        await auth.refreshCodeWhisperer()
+    }
```

Comment:
This will block extension activation. If we _must_ do this on startup, then make it non-blocking:
```suggestion
    if (!isCloud9() && !auth.isConnectionValid()) {
        // this is to proactively check and reflect the state if user's connection is expired
        auth.refreshCodeWhisperer()
    }
```